File: datasets/C++/Qwen/Algorithm/easy5.cpp
#include <iostream>
#include <vector>
using namespace std;

string gameOfStones(int n) {
    // dp[i] represents whether the current player can win with i stones
    if (n <= 1) {
        return "Second";
    }
    
    vector<bool> dp(n + 1, false);
    
    // Fill dp array
    for (int i = 2; i <= n; i++) {
        // Check if we can make a move that leads to opponent's loss
        if (i >= 2 && !dp[i - 2]) {  // Take 2 stones
            dp[i] = true;
        } else if (i >= 3 && !dp[i - 3]) {  // Take 3 stones
            dp[i] = true;
        } else if (i >= 5 && !dp[i - 5]) {  // Take 5 stones
            dp[i] = true;
        }
        // If none of the above, dp[i] remains false
    }
    
    return dp[n] ? "First" : "Second";
}

int main() {
    int t;
    cin >> t;
    
    for (int i = 0; i < t; i++) {
        int n;
        cin >> n;
        cout << gameOfStones(n) << endl;
    }
    
    return 0;
}

File: datasets/Python/Qwen/Algorithm/easy5.py
def gameOfStones(n):
    # dp[i] represents whether the current player can win with i stones
    if n <= 1:
        return "Second"
    
    # Create dp array
    dp = [False] * (n + 1)
    
    # Base cases
    # dp[0] = False (no moves possible)
    # dp[1] = False (no valid moves)
    
    # Fill dp array
    for i in range(2, n + 1):
        # Check if we can make a move that leads to opponent's loss
        if i >= 2 and not dp[i - 2]:  # Take 2 stones
            dp[i] = True
        elif i >= 3 and not dp[i - 3]:  # Take 3 stones
            dp[i] = True
        elif i >= 5 and not dp[i - 5]:  # Take 5 stones
            dp[i] = True
        # If none of the above, dp[i] remains False
    
    return "First" if dp[n] else "Second"

# Read input and solve
t = int(input())
for _ in range(t):
    n = int(input())
    print(gameOfStones(n))